fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QuerySegmentAggregator.java

```diff
@@ -240,6 +240,12 @@ protected SqlQueryData buildSystemLevelQuery(String selectRoot, String subSelect
 
         // Processes through EACH register parameter extracting the integer value
         Map<String, Integer> resourceNameMap = resourceDao.readAllResourceTypeNames();
+        if (resourceNameMap.size() == 0) {
+            // Special condition where we have no registered resources and therefore no data.
+            // This is only used in COUNT
+            queryString.append("SELECT LR.LOGICAL_RESOURCE_ID AS CNT FROM LOGICAL_RESOURCES LR");
```

Comment:
If we know we have no data, should it just be `SELECT 0` or some such?

Reply:
since it's a nested SELECT SELECT it ends up being a pain in the butt to do VALUES() since we don't really want to return a value.

Reply:
yeah, i almost suggested defining and throwing a checked exception so that we don't even attempt the query, but I guess the above is ok.  marking it approved.